src/solidity-frontend/solidity_convert_literals.cpp

```diff
@@ -60,6 +60,24 @@ bool solidity_convertert::convert_bool_literal(
 
 // TODO: Character literal
 // TODO: String literal
+bool solidity_convertert::convert_string_literal(
+  const nlohmann::json &string_literal,
+  std::string the_value,
+  exprt &dest)
+{
+  size_t string_size = the_value.size();
+  typet type = array_typet(
+    signed_char_type(),
+    constant_exprt(
+      integer2binary(string_size, bv_width(int_type())),
+      integer2string(string_size),
+      int_type()));
+
+  string_constantt string(the_value, type);
```

Comment:
Though this ESBMC type has the right name for the usage here, it actually represents a (somewhat broken¹) C-style string literal, that is, it represents the given value with a trailing `\0'. This should probably be fixed. Is Solidity OK with having size+1?

¹ broken because it gets the size in bytes from clang_c_frontend as well as the Solidity one here, and adds +1 to it. For wide-character strings that results in a "size" of the string_constantt of e.g. 21=5*4+1 for the C string literal `L"hello"` if `sizeof(wchar_t) == 4`.

Reply:
You're giving `the_value` to the string_constantt. This will (down the line) be interpreted as the C-style literal `the_value`, including a `\0` at the end. My question: Do string literals in Solidity have a trailing `\0`? If not, does it matter for Solidity? For instance, is it possible to get a specific character from a string by index `str[n]`?

If string literals are supposed to be represented the same way as in C or strings can't be indexed nor their size taken, it's fine. If not, we might need to change `string_constantt`.

Reply:
In solidity strings themselves are not able to be accessed via index, and they need to be cast to a bytes type before their length can be accessed

Reply:
OK, thanks. So when we get to this string -> byte conversion the additional `\0` byte needs to be observed. Could you maybe add a TODO comment here?